.github/workflows/build-and-publish-chart.yml

```diff
@@ -18,7 +18,18 @@ jobs:
       id-token: write
       actions: 'read'
     runs-on: ubuntu-latest
+    env:
+      RELEASE_BRANCH: ''
+
     steps:
+      # Set RELEASE_BRANCH based on the current branch
+      - name: Set RELEASE_BRANCH
+        run: |
+          if [[ "${{ github.ref_name }}" == "main" || "${{ github.ref_name }}" == "develop" ]]; then
+            echo "RELEASE_BRANCH=main" >> $GITHUB_ENV
+          else
+            echo "RELEASE_BRANCH=${{ github.ref_name }}" >> $GITHUB_ENV
+          fi
```

Comment:
we normally release with `github.ref_name` as develop or main. in those cases, this workflow should work exactly the same.